code/modules/overmap/overmap_object.dm

```diff
@@ -128,26 +154,40 @@ var/global/list/overmap_unknown_ids = list()
 	else
 		return TRUE
 
-/obj/effect/overmap/Process()
-	if(!halted && !is_still() && can_move)
-		var/list/deltas = list(0,0)
-		for(var/i = 1 to 2)
-			if(MOVING(speed[i], min_speed))
-				position[i] += speed[i] * OVERMAP_SPEED_CONSTANT
-				if(position[i] < 0)
-					deltas[i] = CEILING(position[i])
-				else if(position[i] > 0)
-					deltas[i] = FLOOR(position[i])
-				if(deltas[i] != 0)
-					position[i] -= deltas[i]
-					position[i] += (deltas[i] > 0) ? -1 : 1
+/obj/effect/overmap/proc/ProcessOvermap(wait, tick)
 
-		update_icon()
+	if(halted || is_still())
+		return PROCESS_KILL
+
+	if(!can_move)
+		return
+
+	var/moved = FALSE
+	var/list/deltas = list(0,0)
+	for(var/i = 1 to 2)
+		if(!MOVING(speed[i], min_speed))
+			continue
+		// Add speed to this dimension of our position.
+		position[i] += Clamp((speed[i] * OVERMAP_SPEED_CONSTANT) * (wait / (1 SECOND)), -1, 1)
+		if(position[i] < 0)
+			deltas[i] = CEILING(position[i])
+		else if(position[i] > 0)
+			deltas[i] = FLOOR(position[i])
+		moved = TRUE
+		// Delta over 0 means we've moved a turf, so we adjust our position accordingly.
+		if(deltas[i] != 0)
+			position[i] -= deltas[i]
+			// Note for future self when confused: this line offsets the effect within the new turf.
+			// Can probably be tidied up at some point but math is spooky.
+			position[i] += (deltas[i] > 0) ? -1 : 1
+
+	if(moved)
 		var/turf/newloc = locate(x + deltas[1], y + deltas[2], z)
 		if(newloc && loc != newloc)
 			Move(newloc)
 			handle_wraparound()
-	handle_overmap_pixel_movement()
+		update_icon()
```

Comment:
Is the reason you are doing this that the underlay being applied here has its scaling transform overridden by the pixel shifts? If so, can you use `KEEP_TOGETHER` or some other method instead?